composer.json

```diff
@@ -12,7 +12,7 @@
         }
     ],
     "require": {
-        "php"                   : ">=5.5",
+        "php"                   : "^5.5.9",
```

Comment:
No, it won't, because you're specifying it in `config`.`platform`, see https://getcomposer.org/doc/06-config.md#platform :)